File: src/main/java/no/nav/bidrag/beregn/saertilskudd/periode/SaertilskuddPeriode.java
package no.nav.bidrag.beregn.saertilskudd.periode;

import java.util.List;
import no.nav.bidrag.beregn.felles.bo.Avvik;
import no.nav.bidrag.beregn.saertilskudd.beregning.SaertilskuddBeregning;
import no.nav.bidrag.beregn.saertilskudd.bo.BeregnSaertilskuddGrunnlag;
import no.nav.bidrag.beregn.saertilskudd.bo.BeregnSaertilskuddResultat;

public interface SaertilskuddPeriode {

  BeregnSaertilskuddResultat beregnPerioder(BeregnSaertilskuddGrunnlag beregnSaertilskuddGrunnlag);

  List<Avvik> validerInput(BeregnSaertilskuddGrunnlag beregnSaertilskuddGrunnlag);

  static SaertilskuddPeriode getInstance() {
    return new SaertilskuddPeriodeImpl(SaertilskuddBeregning.getInstance());
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/saertilskudd/periode/SaertilskuddPeriode.kt
package no.nav.bidrag.beregn.saertilskudd.periode

import no.nav.bidrag.beregn.felles.bo.Avvik
import no.nav.bidrag.beregn.saertilskudd.beregning.SaertilskuddBeregning
import no.nav.bidrag.beregn.saertilskudd.bo.BeregnSaertilskuddGrunnlag
import no.nav.bidrag.beregn.saertilskudd.bo.BeregnSaertilskuddResultat

interface SaertilskuddPeriode {

    fun beregnPerioder(grunnlag: BeregnSaertilskuddGrunnlag): BeregnSaertilskuddResultat

    fun validerInput(grunnlag: BeregnSaertilskuddGrunnlag): List<Avvik>

    companion object {
        fun getInstance(): SaertilskuddPeriode = SaertilskuddPeriodeImpl(SaertilskuddBeregning.getInstance())
    }
}
